services/api/db/views.py

```diff
@@ -1,3 +1,108 @@
-from django.shortcuts import render
+from rest_framework.decorators import action
+from rest_framework.viewsets import ViewSet
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.permissions import IsAdminUser
+from db.models import Profile, OrganizationSettings
+from db.userSerializers import ProfileSerializer, OrganizationSettingsSerializer
+from db.renderer import UserRenderer
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.exceptions import NotFound, ValidationError
 
-# Create your views here.
+
+
+class ProfileViewSet(ViewSet):
+    serializer_class = ProfileSerializer
+    renderer_classes = [UserRenderer]
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        try: 
+            user_profile =  Profile.objects.filter(user=self.request.user)
+            return user_profile
+        except NotFound as e:
+            return Response({'error': e.detail}, status=status.HTTP_404_NOT_FOUND)
+
+    def update_profile_field(self, request, field_name):
+        try:
+            profile = self.get_queryset().first()   #This query returns 404 not found error on failing as provided in get_queryset method.
+            field_value = getattr(profile, field_name)
+            if request.method == 'GET':
+                return Response(field_value, status=status.HTTP_200_OK)
+            elif request.method in ['POST', 'PUT']:
+                try:
+                    new_field_value = request.data.get(field_name)
+                except ValidationError:
+                    return Response({"error" : f"{field_name.capitalize()} field is required"}, status=status.HTTP_400_BAD_REQUEST)
+                setattr(profile, field_name, new_field_value)
+                profile.save()
+                return Response(f"{field_name.capitalize()} updated successfully", status=status.HTTP_200_OK)
+            elif request.method == 'DELETE':
```

Comment:
are we going to allow delete operation for profile fields or just update should be allowed?

Reply:
resolved it